File: Java_projet_perso_/src/main/java/Simpson.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.PrintWriter;
import java.util.concurrent.TimeUnit;

public class Simpson extends Traitement{
	public float pointA;
	public float pointB;
	public float pas;
	public float resultat;
	
	public void setFonction(Fonction fx) {
		this.f = fx;
	}

	public void setPointA(float i) {
		this.pointA = i;
	}

	public void setPointB(float i) {
		this.pointB = i;
	}

	public void setPas(float i) {
		this.pas = i;
	}	
	
	public Simpson() {
		
	}

	public void resolution(String path) throws FileNotFoundException {
		try {
			this.methode="Simpson";		
			PrintWriter fd = new PrintWriter(new File(path));
			float r=0;
			float a=this.pointA,b=this.pointB;
			float h=(Math.abs(b)-Math.abs(a))/this.pas;
			
			for(int i=2;i<this.pas;i+=2){
				r+=2*f.f(a);
				a+=2*h;
				fd.println(a+" "+f.f(a));
			}
			for(int i=1;i<this.pas;i+=2){
				r+=4*f.f(a);
				a+=2*h;
			}
			fd.close();
			this.pathDatas=path;
			this.resultat=(h/3)*r;		
		}catch(Exception e) {
			e.getStackTrace();
			this.pathDatas="null";
		}
	}
}

File: codeigniter/app/Libraries/Simpson.php
<?php
namespace App\Libraries;

class Simpson extends Traitement {
    public $pointA;
    public $pointB;
    public $pas;
    public $resultat;

    public function __construct() {
      
    }  
    
    public function setPointA(float $i) {
      $this->pointA = $i;
    }
  
    public function setPointB(float $i) {
      $this->pointB = $i;
    }
  
    public function setPas(float $i) {
      $this->pas = $i;
    }
  
    public function resolution(string $path) {
      $this->methode = "Simpson";
      try {
          $outputFile = fopen($path, "w");
          if (!$outputFile) {
            throw new Exception("Failed to open file: " . $path);
          }
        $r = 0;
        $a = min($this->pointA, $this->pointB);
        $b = max($this->pointA, $this->pointB);
        $h =(float) (($b - $a) / $this->pas);
  
        // Even terms (multiples of 2)
        for ($i = 2; $i < $this->pas; $i += 2) {
          $r +=(float) (2 * $this->f->f($a + $i * $h));
          fwrite($outputFile, ($a + $i * $h) . " " . $this->f->f($a + $i * $h) . PHP_EOL);
        }
  
        // Odd terms (multiples of 1)
        for ($i = 1; $i < $this->pas; $i += 2) {
          $r +=(float) (4 * $this->f->f($a + $i * $h));
          fwrite($outputFile, ($a + $i * $h) . " " . $this->f->f($a + $i * $h) . PHP_EOL);
        }
  
        $this->resultat =(float) (($h / 3) * $r);
        fclose($outputFile);
        $this->pathDatas = $path;
      } catch (Exception $e) {
        echo $e->getMessage() . PHP_EOL;
        $this->pathDatas = "null";
      }
    }
  }
?>